lms/djangoapps/program_enrollments/api/writing.py

```diff
@@ -424,3 +449,52 @@ def _organize_requests_by_external_key(enrollment_requests):
             continue
         requests_by_key[key] = request
     return requests_by_key, duplicated_keys
+
+
+def _process_conflicted_course_enrollments(
+    requests_by_key,
+    existing_course_enrollments,
+    program_uuid,
+    course_key
+):
+    """
+    Process the list of existing course enrollments together with
+    the enrollment request list stored in 'requests_by_key'. Detect
+    whether we have conflicted ACTIVE ProgramCourseEnrollment entries.
+    When detected, log about it and set that update or create entry to
+    have duplicated status.
+
+    Arguments:
+        requests_by_key (dict)
+        existing_course_enrollments (queryset),
+        program_uuid (UUID|str),
+        course_key (str)
+
+    Returns:
+        results (dict) with detected conflict entry, or empty dict.
+    """
+    results = {}
+
```

Comment:
can you rename some of these variables so that the role they play in this function is more clear? e.g.:
`course_enrollment_statuses_by_user_key` -> `requested_statuses_by_user_key`
`enrollment_status` -> `requested_status`
`course_enrollment` -> `existing_enrollment`